releases.md

```diff
@@ -0,0 +1,103 @@
+# Tekton Release Cycles
+
+## Support Policy
+
+The Tekton project maintains three release branches for each project, created one per quarter, which 
+results in a overall support window of approximately one year for each of these releases.
+
+This approach allows Tekton to roughly align to the [Kubernetes release cycles][kube-releases] and 
+maintain three supported releases at any point in time.
+
+Tekton is made of a collection of project, and each project may define its own release cadence, as long as
+this [support policy](#support-policy) is followed. 
+
+Examples:
+
+- A project may release on a monthly basis, and maintain a release branch for every fourth release
+- A project may release on a quarterly basis, and maintain a release branch for the three most recent minor 
+  releases
+
+## Release Numbers
+
+Tekton uses [semantic versioning][semantic] to version its own releases. Release numbers are in the format
+MAJOR.MINOR.PATCH `vX.Y`. The [support policy](#support-policy) applies to MAJOR and MINOR releases alike.
+
+## Release Tags and Branches
+
+Every time a Tekton project produces a release, a new tag `vX.Y.Z` is created, as well as a new branch that
+initially points to the git tag `vX.Y.Z`.
+
+- Supported releases: 
+    - branch name: `release-vX.N-lts`
+    - support window: until `release-vX.N+3`
+
+- Any other major or minor release:
+    - branch name: `release-vX.M`
+    - support window: until `release-vX.M+1`
+
+Once a release support window expires, the corresponding branch is deleted.
+It is recommended (but not mandatory) to make major releases coincide with supported ones.
+
+## Milestones
+
+Tekton projects may use [GitHub milestones][github-milestones] to plan their next release. This helps the
+maintainer and reviewer team to focus their work for the release and it gives users visibility on what they
+may expect for the upcoming releases.
+
+When used, milestone names must include the release number in `v<MAJOR>.<MINOR>` format.
+Once a release is made the milestone is closed. Patch releases are not tracked as part of the milestone.
+
+## Project Requirements
+
+Project must include release documentation in a `releases.md` file, which must include:
+
+- The release frequency adopted by the project
+- The first release where this [support policy](#support-policy) applies
+- A list of one year worth of releases. For each release:
```

Comment:
Wouldn't it be less work to add the EOL for a release to the Release note and then point users to the Github releases page for a project?

Reply:
I was hoping to automate this from the GitHub release page somehow - I'd like to check how https://kubernetes.io/releases/ is produced and get something similar if possible.

If not automated this can be built incrementally, like what we have in https://github.com/tektoncd/pipeline#read-the-docs